lib/OpenQA/Worker/Settings.pm

```diff
@@ -71,6 +72,30 @@ sub new ($class, $instance_number = undef, $cli_options = {}) {
     return $self;
 }
 
+sub _is_section_relevant ($section_name, $instance_number) {
+    return 1 if $section_name eq $instance_number;
+    return 0 unless looks_like_number $instance_number;
```

Comment:
`looks_like_number` also returns true for floats and Infinity. Maybe we can be a bit more strict here?